firewood/src/merkle/stream.rs

```diff
@@ -470,6 +420,40 @@ mod tests {
         check_stream_is_done(merkle.iter(root)).await;
     }
 
+    #[tokio::test]
+    async fn no_start_key() {
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        for i in (0..256).rev() {
+            for j in (0..256).rev() {
+                let key = vec![i as u8, j as u8];
```

Comment:
```suggestion
        for i in 0..=u8::MAX {
            for j in 0..=u8::MAX {
                let key = vec![i, j];
```
Why do you care about insertion order?

Reply:
Added a comment -- this is just to ensure that the iterator doesn't somehow return key-value pairs in their insertion order, although I suppose this isn't really necessary.